disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -35,6 +38,16 @@ maybeGen g = sized $ \s ->
     (1, return Nothing),
     (s, Just <$> resize (s `div` 2) g)]
 
+-- | Wait for a generated `Just` value
+--
+-- Use _only_ in case of emergencies when you have no other way to get an `a` safely
+genFromMaybe :: Gen (Maybe a) -> Gen a
+genFromMaybe g =
+  suchThat g isJust >>= \ma ->
+    case ma of
+      Just a -> pure a
+      Nothing -> fail "Failed to generate a Just"
```

Comment:
Sorry to nitpick, maybe add "Disorder.Core.Gen.genFromMaybe: " as a prefix so it's easy to tell where the error comes from? I've been bitten recently by this and resorted to searching for error strings in `base` to find out what was going on :(